src/virtual_ptdf_calculations.jl

```diff
@@ -1,4 +1,74 @@
-# TODO: add descriptions to functions
+struct RowCache
+    temp_cache::Dict{Int, Array{Float64}}
+    persistent_cache_keys::Set{Int}
+    max_cache_size::Int
+    max_num_keys::Int
+end
+
+function RowCache(max_cache_size::Int, persistent_rows::Set{Int}, row_size)
+    if (length(persistent_rows) + 1) * row_size > ROW_PERSISTENT_CACHE_WARN
+        @warn "The minimum cache size for the persisted row is over 1 GiB. This specification will cause large memory usage"
```

Comment:
If this condition is true then max_cache_size is effectively ignored. Should you reset it here?